src/main.cpp

```diff
@@ -86,6 +100,43 @@ Entity target{Vec2<float>{randomFloat(padding_v, window_width_v - padding_v),
 
 void playerUpdate(float delta)
 {
+  if (gInput.heldKeys.contains(SDLK_w))
+  {
+    playerDirection.y = -1;
```

Comment:
Refactor this logic to _accumulate_ input instead of assigning: `playerDirection.y -= 1.0f;`. You won't have to handle left + right etc separately then.